src/main/java/org/ibp/api/java/impl/middleware/study/StudyServiceImpl.java

```diff
@@ -381,4 +382,26 @@ public void updateStudy(final Study study) {
 		this.studyValidator.validate(studyId, false);
 		this.studyDataManager.updateStudyLockedStatus(studyId, study.isLocked());
 	}
+
+	@Override
+	public long countVariablesByDatasetId(final int studyDbId, final List<Integer> variableTypes) {
+		return this.middlewareStudyService.countVariablesByDatasetId(studyDbId, variableTypes);
+	}
+
+	@Override
+	public List<VariableDTO> getVariablesByDatasetId(final int pageSize, final int pageNumber, final int studyDbId,
+		final String cropname, final List<Integer> variableTypes) {
+		return this.middlewareStudyService.getVariablesByDatasetId(pageSize, pageNumber, studyDbId, variableTypes, cropname);
```

Comment:
`cropName `is not essential for Middleware querying hence I don't think there's a need to pass to as parameter to Middleware service. We can just set `cropName `in BMSAPI service code.